kapitan/utils.py

```diff
@@ -471,3 +477,95 @@ def make_request(source):
     else:
         r.raise_for_status()
     return None, None
+
+
+def unpack_downloaded_file(file_path, output_path, content_type):
+    """unpacks files of various MIME type and stores it to the output_path"""
+    if content_type == "application/x-tar":
+        tar = tarfile.open(file_path)
+        tar.extractall(path=output_path)
+        tar.close()
+        is_unpacked = True
+    elif content_type == "application/zip":
+        zfile = ZipFile(file_path)
+        zfile.extractall(output_path)
+        zfile.close()
+        is_unpacked = True
+    elif content_type in ["application/octet-stream", "application/x-gzip"]:
+        if re.search(r"(\.tar\.gz|\.tgz)$", file_path):
+            tar = tarfile.open(file_path)
+            tar.extractall(path=output_path)
+            tar.close()
+            is_unpacked = True
+        else:
+            extension = re.findall(r"\..*$", file_path)[0]
+            logger.debug(f"file extension {extension} not suported")
+            is_unpacked = False
+    else:
+        logger.warning(f"content type {content_type} not suported")
+        is_unpacked = False
+    return is_unpacked
+
+
+def safe_copy_file(src, dst):
+    """Copy a file from 'src' to 'dst'.
+
+    Similar to distutils.file_util.copy_file except
+    if the file exists in 'dst' it's not clobbered
+    or overwritten.
+
+    returns a tupple (src, val)
+    file not copied if val = 0 else 1
+    """
+
+    if not os.path.isfile(src):
+        raise DistutilsFileError("can't copy {}: doesn't exist or is not a regular file".format(src))
+
+    if os.path.isdir(dst):
+        dir = dst
```

Comment:
The `dir` variable in this function seems to be unused. Why assign values to it?

Reply:
Thanks for pointing it out, the `dir` variable was to add a debugging message after copying the file. I'll add it.